### DISCUSSION TITLE
Implies function

### DISCUSSION THREAD

**Discussion by VivonaLopez** [yalmip/YALMIP#1100] [Optimization modelling] [answered]
In my optimization problem, I need to implement an if statement which depends on a sdpvar variable. I have read and tried to solve it using the implies function; however, the model is not getting the behavior that I expected.
I am writing an optimization model which maximizes the power injection in an electrical network using some constraints.
I am trying to define this constraint:
If 0.95<Vm<1.05, then Qcontrol >0
If Vm = 1, then Qcontrol ==0
If 1<Vm<1.05, then Qcontrol <0

Where "Vm" and Q_control are sdpvar variables, Q_control is limited by a real number Q_available.
 With this constraint, I am trying to "force" the system to inject or consume Q_control based on the statement above.
I did it using these lines:
for t=1:T
    for bus=1:length(RES)
        C=[C,-Q_available(t)<=Q_control(RES(bus),t)<=Q_available(t)];
        C = [C,[sum(d) == 1,
        implies([0.95<=Vm(RES(bus),t) <= 1, Q_control(RES(bus),t)>=0],d(1));
        implies([Vm(RES(bus),t) == 1,  Q_control(RES(bus),t)==0],d(2));
        implies([1<=Vm(RES(bus),t) <= 1.05,  Q_control(RES(bus),t)<=0],d(3))]];
        end
end
Even I tried using the syntax with d at the beginning 
for t=1:T
    for bus=1:length(RES)
     C=[C,-Q_available(t)<=Q_control(RES(bus),t)<=Q_available(t)];
        C = [C,[sum(d) == 1,
        implies(d(1),[0.95<=Vm(RES(bus),t) <= 1, Q_control(RES(bus),t)>=0]);
        implies(d(2),[Vm(RES(bus),t) == 1,  Q_control(RES(bus),t)==0]);
        implies(d(3),[1<=Vm(RES(bus),t) <= 1.05,  Q_control(RES(bus),t)<=0])]];
    end
end

But I am not getting the results that I need. I would like to ask you if you can advise me which syntax is the correct option and if I am missing any critical step in this problem.
I appreciate your help.

**Comment by johanlofberg** [comment, marked as answer]
You cannot use the same variable d for all t and bus.

**Reply by VivonaLopez** [reply]
Thank you for your reply, Professor. 
I am new in yalmip, and I am struggling with my models.
I have another question, 
I am trying to multiply two sdpvar variables, 
bin_V, and Q_Control
((value(bin_V(RES(bus),t))))*value(Q_control(RES(bus),t));
but I am getting the result " NaN; I checked, and each variable has a result, but I am not sure why I got this error when I try to operate them.
Thank you for your help.

**Reply by johanlofberg** [reply]
To multiply variables x and y, you simply write x*y as always. value makes no sense here, as it simply extracts the computed solution after solving a problem.

However, it sounds like one variable is binary, and then it is bad modelling to use a bilinear product, as you can model it using linear constraints via implies 

https://yalmip.github.io/example/unitcommitment/
https://yalmip.github.io/tutorial/logicprogramming
